diff --git a/Dockerfile.dapper b/Dockerfile.dapper
index d2e8c93..69d6980 100644
--- a/Dockerfile.dapper
+++ b/Dockerfile.dapper
@@ -2,7 +2,7 @@ ARG BASE_BRANCH
 FROM quay.io/submariner/shipyard-dapper-base:${BASE_BRANCH}
 
 ARG PROJECT
-ENV DAPPER_ENV="QUAY_USERNAME QUAY_PASSWORD CLUSTERS_ARGS DEPLOY_ARGS CLEANUP_ARGS E2E_ARGS RELEASE_ARGS MAKEFLAGS FOCUS SKIP PLUGIN E2E_TESTDIR GITHUB_USER GITHUB_TOKEN" \
+ENV DAPPER_ENV="QUAY_USERNAME QUAY_PASSWORD CLUSTERS_ARGS DEPLOY_ARGS CLEANUP_ARGS E2E_ARGS RELEASE_ARGS MAKEFLAGS FOCUS SKIP PLUGIN E2E_TESTDIR GITHUB_USER GITHUB_TOKEN DOCKER_USER DOCKER_PASSWD" \
     DAPPER_SOURCE=/go/src/github.com/submariner-io/${PROJECT} DAPPER_DOCKER_SOCKET=true
 ENV DAPPER_OUTPUT=${DAPPER_SOURCE}/output
 
diff --git a/scripts/shared/clusters.sh b/scripts/shared/clusters.sh
index 28547e9..4995922 100755
--- a/scripts/shared/clusters.sh
+++ b/scripts/shared/clusters.sh
@@ -44,6 +44,27 @@ set -em
 source ${SCRIPTS_DIR}/lib/debug_functions
 source ${SCRIPTS_DIR}/lib/utils
 
+function docker_login() {
+    login="$1"
+    passwdfile="$2"
+
+    if [ ! -z "${login}" ] && [ ! -z "${passwdfile}" ] ; then
+        cat ${passwdfile} | docker login -u ${login} --password-stdin &>/dev/null
+        if [ $? == 0 ] ; then
+            echo "Docker login with user ${login} successful"
+        else
+            echo "Docker login with user ${login} failed"
+        fi
+    fi
+}
+
+DOCKER_PASSWD_FILE="${DAPPER_OUTPUT}/.docker_file"
+echo "${DOCKER_PASSWD}" > "${DOCKER_PASSWD_FILE}"
+docker_login ${DOCKER_USER} ${DOCKER_PASSWD_FILE}
+rm -f "${DOCKER_PASSWD_FILE}"
+unset DOCKER_PASSWD
+unset DOCKER_PASSWD_FILE
+
 ### Functions ###
 
 function generate_cluster_yaml() {
